.github/workflows/check-and-publish.yaml

```diff
@@ -59,6 +59,25 @@ jobs:
         uses: pypa/gh-action-pypi-publish@release/v1
         with:
           repository-url: https://test.pypi.org/legacy/
+
+  publish-production:
+    name: Publish
+    if: ${{ github.event_name == 'push' && vars.PUBLISH_PYPI == 'true' && (startsWith(github.ref, 'refs/tags') || github.ref == 'refs/heads/master') }}
```

Comment:
This only needs to run for tags, not normal commits on master. With that change, the `if` in the publish step can be removed.

Reply:
Thanks for catching that. I've updated my branch.